crashreports/crashreporter/CrashReportApp.cpp

```diff
@@ -0,0 +1,464 @@
+                ++i;
+            }
+            else
+                throw std::logic_error("malformed parameters string");
+        }
+        if (state != TOKEN_VALUE)
+            throw std::logic_error("malformed parameters string");
+
+        return result;
+    }
+
+    void PrintMinidump(google_breakpad::Minidump& minidump)
+    {
+        google_breakpad::BasicSourceLineResolver resolver;
+        google_breakpad::MinidumpProcessor minidumpProcessor(nullptr, &resolver);
+        google_breakpad::MinidumpThreadList::set_max_threads(std::numeric_limits<uint32_t>::max());
+        google_breakpad::MinidumpMemoryList::set_max_regions(std::numeric_limits<uint32_t>::max());
+        
+        google_breakpad::ProcessState processState;
+        
+        if (minidumpProcessor.Process(&minidump, &processState) != google_breakpad::PROCESS_OK)
+        {
+            printf("Failed to process minidump");
+        }
+        else
+        {
+            google_breakpad::PrintProcessState(processState, true, &resolver);
+        }
+    }
+
+    wxString MakeDumpString(google_breakpad::Minidump& minidump, const wxString& temp)
+    {
+#if _WIN32
+        auto stream = _wfreopen(temp.wc_str(), L"w+", stdout);
+#else
+        auto stream = freopen(temp.utf8_str().data(), "w+", stdout);
+#endif
+        if (stream == NULL)
+            throw std::runtime_error("Failed to print minidump: cannot open temp file");
+        PrintMinidump(minidump);
+        fflush(stdout);
+
+        auto length = ftell(stream);
+        std::vector<char> bytes(length);
+        fseek(stream, 0, SEEK_SET);
+        fread(&bytes[0], 1, length, stream);
+        fclose(stream);
+        
+#if _WIN32
+        _wremove(temp.wc_str());
+#else
+        remove(temp.utf8_str().data());
+#endif
+
+        return wxString::From8BitData(&bytes[0], bytes.size());
+    }
+
+    wxString MakeHeaderString(google_breakpad::Minidump& minidump)
+    {
+        if (auto exception = minidump.GetException())
+        {
+            if (auto rawException = exception->exception())
+            {
+                // i18n-hint C++ programming assertion
+                return wxString::Format(_("Exception code 0x%x"), rawException->exception_record.exception_code);
+            }
+            else
+            {
+                // i18n-hint C++ programming assertion
+                return _("Unknown exception");
+            }
+        }
+        else if (auto assertion = minidump.GetAssertion())
+        {
+            auto expression = assertion->expression();
+            if (!expression.empty())
+            {
+                return expression;
+            }
+        }
+        return _("Unknown error");
+    }
+
+    void DoShowCrashReportFrame(const wxString& header, const wxString& dump, const std::function<bool(const wxString& comment)>& onSend)
+    {
+        static constexpr int MaxUserCommentLength = 2000;
+
+        auto frame = new wxFrame(
```

Comment:
There are many "naked news" in this file.  I won't insist on cleaning them up, in this small, separate, short-running executable.

But you should know about the conventions I insist on in Audacity.

1) When you construct anything derived from wxWindow, and its parent is null, then that may "leak" a window that gets cleaned up only when the application exits.  (But modal dialogs may have null parent windows, because they are stack allocated.)